### PR TITLE
[EventGrid] Ignite Release generate with new typespec

### PR DESCRIPTION
**Pull request by l0lawrence**
ToDo to merge:

- [x] Test new features
- [x] Add sample cases for new features

Will be doing a separate PR for documentation and release updates (changelog etc) b/c also need to release system events this month as well ( most likely GA release will be -- 4.16 and Beta -- 4.17.0b1)

### CODE REVIEW

File: sdk/eventgrid/tests.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ stages:
     parameters:
       ServiceDirectory: eventgrid
       BuildTargetingString: azure-eventgrid*
+      Location: centraluseuap
```
Comment (l0lawrence) [inline sdk/eventgrid/tests.yml:8]: set location to centraluseuap -- currently only canary supports the mgmt api version for deploying resources -- will update when deployed fully

File: sdk/eventgrid/azure-eventgrid/CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
-## 4.15.1 (Unreleased)
+## 4.16.0b1 (Unreleased)
 
 ### Features Added
 
-- Added a new EventGridClient that supports `publish_cloud_events`, `receive_cloud_events`, `acknowledge_cloud_events` , `release_cloud_events`, and `reject_cloud_events` operations.
-- Added new models to support these new operations on EventGridClient.
+- Added new features to the EventGridClient that supports `publish_cloud_events`, `receive_cloud_events`, `acknowledge_cloud_events` , `release_cloud_events`, and `reject_cloud_events` operations. These features include a `renew_cloud_event_locks` operation, as well as a `release_with_delay` parameter on the `release_cloud_events` operation.
+- Added new models to support these new features on EventGridClient.
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/CHANGELOG.md:8]: does a note need to be made about lock_token -> reject_options/acknowledge_options or no?

**Review by swathipil — APPROVED** [review]

File: sdk/eventgrid/azure-eventgrid/CHANGELOG.md (hunk 2 above)
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/CHANGELOG.md:8]: will add

**Pull request merged**